src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/IProjectSnapshot.cs

```diff
@@ -37,6 +37,8 @@ internal interface IProjectSnapshot
     ProjectWorkspaceState ProjectWorkspaceState { get; }
 
     RazorProjectEngine GetProjectEngine();
+
+    bool ContainsDocument(string filePath);
     IDocumentSnapshot? GetDocument(string filePath);
```

Comment:
💭 should we get rid of `GetDocument` entirely?

Reply:
I thought about that, but it requires a _much_ larger change. This PR doesn't address most of the existing `GetDocument(...)` call-sites. Also, I think the API is probably still useful. I'd rather make it non-nullable and throw at some point -- sort of like `TryGetProject` and `GetLoadedProject` on `IProjectSnapshotManager`.